### DISCUSSION TITLE
YARP behind a lan webproxy

### DISCUSSION THREAD

**Discussion by johanhallgrengiszeal** [dotnet/yarp#1662] [Q&A] [answered]
I have used YARP for a while but now I have a new situation where the application must pass a lan proxy to reach internet. 

Whithout any changes the yarp-app returns Bad Gatway (502)

I have tried to set the defaultProxy setting in web.config and it worked for a regualar ashx-page, but did not change the function in the yarp-app, it just returns Bad Gatway (502). 

I found some information about WebProxy here in the soure code but no example that I understund and no example/information if it could be set in the appsetting-file. 

Any help would be appreciated.

Best ragards
johan

**Comment by MihaZupan** [comment, marked as answer]
Yes, the `WebProxy` can be specified through the configuration file (we've just [missed documenting it](https://github.com/microsoft/reverse-proxy/issues/1672), sorry :)).

The proxy configuration is defined by [`WebProxyConfig`](https://github.com/microsoft/reverse-proxy/blob/main/src/ReverseProxy/Configuration/WebProxyConfig.cs) and goes under the `HttpClient` config:
```json
{
  "Clusters": {
    "cluster1": {
      "HttpClient": {
        "WebProxy": {
          "Address": "http://my-proxy"
        }
      }
    }
  }
}
```

**Reply by johanhallgrengiszeal** [reply]
Prefect, I will try it:)

**Reply by johanhallgrengiszeal** [reply]
It works, after some struggle!

/Johan

**Reply by MihaZupan** [reply]
Hey, can you share what issues you were hitting here before making it work? Is there anything we can do to make it easier in the future?

**Reply by MichelZ** [reply]
Yeah, the struggle is real... :)
I think a flag "UseSystemProxy" or "UseDefaultProxy" would be good... 

Any insight on why you want to be special by ignoring all the "default" .NET proxy stuff?

**Comment by johanhallgrengiszeal** [comment]
I just typed wrong in the appsetting-file☹

But improved documentation would be great!

Thanks
Johan

Johan Hallgren
+46 (0)70 200 41 01
***@***.******@***.***>
GISZEAL AB
www.giszeal.se<http://www.giszeal.se/>
[logo-red]
